src/playback-watcher.js

```diff
@@ -29,35 +29,54 @@ const timerCancelEvents = [
  *        Options object
  * @param {TimeRange} options.buffered
  *        Current buffer
+ * @param {TimeRange} options.videoBuffered
+ *        Current buffer from the media sources video buffer
+  * @param {TimeRange} options.audioBuffered
+  *       Current buffer from the media sources audio buffer
  * @param {number} options.targetDuration
  *        The active playlist's target duration
  * @param {number} options.currentTime
  *        The current time of the player
  * @return {boolean}
  *         Whether the current time should be considered close to the buffer
  */
-export const closeToBufferedContent = ({ buffered, targetDuration, currentTime }) => {
-  if (!buffered.length) {
-    return false;
+export const closeToBufferedContent = ({ buffered, audioBuffered, videoBuffered, targetDuration, currentTime }) => {
+  const twoSegmentDurations = (targetDuration - Ranges.TIME_FUDGE_FACTOR) * 2;
+  const bufferedToCheck = [audioBuffered, videoBuffered];
+
+  for (let i = 0; i < bufferedToCheck.length; i++) {
+    // skip null buffered
+    if (!bufferedToCheck[i]) {
+      continue;
+    }
+
+    const timeAhead = Ranges.timeAheadOf(bufferedToCheck[i], currentTime);
+
+    // if we are less than two video/audio segment durations behind,
+    // we haven't append enough to be close to buffered content.
+    if (timeAhead < twoSegmentDurations) {
+      return false;
+    }
   }
 
-  // At least two to three segments worth of content should be buffered before there's a
-  // full enough buffer to consider taking any actions.
-  if (buffered.end(0) - buffered.start(0) < targetDuration * 2) {
-    return false;
+  // default to using buffered, but if we don't have one
+  // use video or audio buffered
+  if (!buffered || buffered.length === 0) {
+    buffered = videoBuffered || audioBuffered
```

Comment:
oops forgot a semicolon
```suggestion
    buffered = videoBuffered || audioBuffered;
```